### DISCUSSION TITLE
Setting a manual resolution / Refresh

### DISCUSSION THREAD

**Discussion by aemurdoch** [ClassicOldSong/Apollo#161] [General]
Hi, just have this setup and its working great across the 3x clients I currently use (a 4k LG tv, a 2k 16:10 tablet, and a steamdeck oled) - amazing work and love the way it spawns and disables the custom virtual screens.

My question is how/if its possible to set a custom resolution for the steam deck - when it connects it automatically sets its self as 1280x800 (native) as to be expected - there are higher res 16:9 reses I can choose (1920x1080, 2560x1440, 3480x2160 etc).

What im actually after is running the display at 2560x1600 (so its native 16:10 4x res of the steam deck) and just having the stream to the deck run at 1280x800 - only issue is I cannot find a way to manually assign a custom resolution (is there a way to define this?)

If I force the res from the client side (to 2560x1600) it connects and sets this up on the virtual display - but then reverting the stream back to 1280x800 this option vanishes.

**Comment by ClassicOldSong** [comment]
There's custom resolution option on Moonlight Qt, IDK if SteamDeck can configure this. If there's still no option to configure it on the client, there's an option named "Resolution Scale Factor" under each app's settings in Apollo. You can set it to 200% and it'll give you the total 4x resolution you expect.

**Comment by aemurdoch** [comment]
Thanks for the response, the custom resolution is what I tried as it happens to force resolutions (and works) but results in the entire stream being sent at that resolution (so the stream is sent at 4x).

What I was after doing, was actually to set the desktop resolution at 4x (so games can run at that native res) and the stream set natively (so essentially its downsampling the 4x stream - bit like super sampling).

Your advice helped though - I created a 2nd duplicate APP (so playnite for example) using 200% res just for use with the SD - if you open this it sets the desktop res at the required 2560x1600, allows the desktop (and game to run at this res) but still streams this as native (and you can set the icon size etc in windows at 200% that way it all appears at a nomal sizing).

Its a bit of a clunky way to do it, but works and means that at least from my POV it still renders games at the highest quality (also means I dont have to keep changing res in game from 1280x800 to 2560x1440 - which is native on all my other devices including my phyisical monitor).

**Reply by ClassicOldSong** [reply]
With Artemis you can actually set resolution scale factor client side, but unfortunately that's not available on the upstream qt version.

**Reply by Skyyblaze** [reply]
I'm a new Apollo user and first of all I want to thank you for your amazing work @ClassicOldSong, the Virtual Display feature makes streaming from my Windows PC to my ROG Ally so much more convenient! And all the other features like the Permission System are also a fantastic idea. 

However like @aemurdoch I would also like to request being able to set the resolution of the Virtual Display independently from the actual stream resolution. I normally run a 4k display so I would like the Virtual Display to always be in 4k because switching desktop resolutions can sometimes lead to a messy UI in Windows and some games like Chrono Cross also fail to check for the new desktop resolution and simply start out of bounds having to set the resolution back and forth each time I switch devices.

So having the Virtual Display always at 4k would be preferred by me though I don't need the actual stream to be running at 4k. Maybe it could be host-side option since you can already set a Fallback Resolution? Something like: "Always Enforce Fallback Resolution for Virtual Display" could be a possibility.

Meanwhile I found an easy  workaround. I start the stream with Moonlight set to 4k and then I back out of the stream without exiting the Virtual Desktop app, set Moonlight to 720p and then reconnect to the session.

I look forward to Artemis being available on Windows and again, thanks for your great work! :)
